File: e417/e417.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int a;
    while (cin >> a) {
        long long int num1 = 0, num2 = 0;

        for (int i = 0; i < a; i++) {
            int x;
            cin >> x;
            num1 += x;
            num2 += 1LL * x * x;
        }

        long long int ans = (num1 * num1 - num2) / 2;
        cout << ans << endl;
    }
    return 0;
}

File: e417-乘法~乘法~加法~/e417.java
import java.util.Scanner;

public class e417 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        while (scanner.hasNextInt()) {
            int a = scanner.nextInt();

            long num1 = 0, num2 = 0;

            for (int i = 0; i < a; i++) {
                int x = scanner.nextInt();
                num1 += x;
                num2 += (long) x * x;
            }

            long ans = (num1 * num1 - num2) / 2;
            System.out.println(ans);
        }

        scanner.close();
    }
}